pymeasure/instruments/agilent/agilentE4407B.py

```diff
@@ -0,0 +1,637 @@
+
+    # sweep commands
+    frequency_points = Instrument.control(
+        ":SENSe:SWEEp:POINts?",
+        ":SENSe:SWEEp:POINts %g",
+        """ An integer property that represents the number of frequency
+        points in the sweep. This property can take values from 101 to 8192.
+        """,
+        validator=truncated_range,
+        values=[101, 8192],
+        cast=int,
+    )
+    sweep_time = Instrument.control(
+        ":SENS:SWE:TIME?",
+        ":SENS:SWE:TIME %g",
+        """ A floating point property that represents the sweep time
+        in seconds. This property can be set.
+        """,
+    )
+    number_of_segments = Instrument.measurement(
+        ":SENS:SWEep:SEGMent:COUNT?",
+        """ An integer property that represents the number of segments
+        in the sweep. This property is read-only.
+        """,
+    )
+    set_all_segments_sst = Instrument.control(
+        ":SENS:SWE:SEGM:DATA? SST",
+        ":SENS:SWE:SEGM:DATA SST,%g",
+        """ A command that sets all the segments of a sweep, at once, with a string.
+        format is start, stop, rbw, vbw, points, time
+        """,
+    )
+    set_all_segments_csp = Instrument.control(
+        ":SENS:SWE:SEGM:DATA? CSP",
+        ":SENS:SWE:SEGM:DATA CSP,%g",
+        """ A command that sets all the segments of a sweep, at once, with a string.
+        format is center, span, rbw, vbw, points, time
+        """,
+    )
+    merge_segments_sst = Instrument.setting(
+        ":SENS:SWE:SEGM:DATA:MERge SST %g",
+        """
+        A command that merges the data with current the segments of a sweep with a string.
+        format is start, stop, rbw, vbw, points, time
+        """,
+    )
+    merge_segments_csp = Instrument.setting(
+        ":SENS:SWE:SEGM:DATA:MERge CSP %g",
+        """
+        A command that merges the data with current the segments of a sweep with a string.
+        format is center, span, rbw, vbw, points, time
+        """,
+    )
+
+    def delete_segments(self):
+        """Deletes all the segmented sweep data."""
+        self.write(":SENS:SWE:SEGM:DEL")
+
+    delete_segment = Instrument.setting(
+        ":SENS:SWE:SEGM:DEL %g",
+        """
+        Deletes the specifed segment of a sweep.""",
+    )
+
+    # Sensor commands
+    resolution_bandwidth = Instrument.control(
+        ":SENS:BAND:RES?",
+        ":SENS:BAND:RES %g",
+        """ A floating point property that represents the resolution bandwidth
+        in Hz. This property can be set.
+        """,
+    )
+    video_bandwidth = Instrument.control(
+        ":SENS:BAND:VID?",
+        ":SENS:BAND:VID %g",
+        """ A floating point property that represents the video bandwidth
+        in Hz. This property can be set.
+        """,
+    )
+    resolution_bandwidth_auto = Instrument.control(
+        ":SENS:BAND:RES:AUTO?",
+        ":SENS:BAND:RES:AUTO %g",
+        """ A boolean property that represents the resolution bandwidth
```

Comment:
For boolean properties, you can use the pattern
```
map_values=True,
values={True: 1, False: 0},  # or as appropriate for your device
```
then you can use Python booleans to interact with this.

A good name for boolean properties can be understood as a yes/no question (`output_enabled`) -- here maybe `resolution_bandwidth_auto_enabled`.
Similar below.